docs/csharp/language-reference/compiler-messages/cs8410.md

```diff
@@ -0,0 +1,34 @@
+---
+title: Compiler Error CS8410
+ms.date: 07/11/2020
+f1_keywords:
+  - "CS8410"
+helpviewer_keywords:
+  - "CS8410"
+author: Youssef1313
+---
+# Compiler Error CS8410
+
+'type': type used in an asynchronous using statement must be implicitly convertible to 'System.IAsyncDisposable' or implement a suitable 'DisposeAsync' method.
+
+The expression inside an `await using` statement must have a `DisposeAsync` method.
+
+## To correct this error
+
+Use `using` instead of `await using`.
+
+## Example
+
+<!-- TODO: The current example is a little bit poor. -->
+
+```csharp
+class C
+{
+    public static async System.Threading.Tasks.Task Main()
+    {
+        await using (Main)
+        {
+        }
+    }
+}
+```
```

Comment:
Consider adding a see also and linking to the Implement DisposeAsync article.

```markdown
## See also

- [Implement a DisposeAsync method](../../../standard/garbage-collection/implementing-disposeasync.md)
```

Reply:
Sure. I even opened https://github.com/dotnet/docs/issues/19440 specifically for this. But looks like I forgot 😄